### DISCUSSION TITLE
Any way to self-throttle downloading?

### DISCUSSION THREAD

**Discussion by toydotgame** [Tyrrrz/DiscordChatExporter#1034] [Q&A] [answered]
I'd imagine when doing an `exportall` and using `--media` would cause a tremendous amount of `GET` requests or something similar to the Discord servers. Speed isn't a big requirement for me because I just want a general archive of the last 4 years I've spent on Discord, but I have sent nearly 90k messages and likely have access to around 700k-ish messages throughout all my servers and DMs.

This would be an insane amount of data to request at these inhuman speeds from the servers, and I feel like I might get flagged for this. I don't want to lose my account so I'm wondering if there's a way to impose a rate limit on myself that's more at the speed of "some insane person who decided to go back and read the entire channel history," potentially over the course of a week or two, rather than requesting it all as fast as possible over a few hours (or however long it takes).

**Comment by Tyrrrz** [comment]
There's no need, the app respects rate limits

**Reply by toydotgame** [reply]
How fast are the rate limits? I'm pretty paranoid about being flagged for anything that would seem unusual. I might try downloading a few channels from big public servers on an alt to calm my nerves.

**Reply by Tyrrrz** [reply, marked as answer]
It's based on response headers: https://github.com/Tyrrrz/DiscordChatExporter/blob/c945c98f75480699e3c74e56aef250641128b4ba/DiscordChatExporter.Core/Discord/DiscordClient.cs#L54-L83

Also: https://github.com/Tyrrrz/DiscordChatExporter/blob/c945c98f75480699e3c74e56aef250641128b4ba/DiscordChatExporter.Core/Utils/Http.cs#L42-L50

https://discord.com/developers/docs/topics/rate-limits

**Reply by toydotgame** [reply]
Thanks! Hopefully rate limits are all that Discord will do. It doesn't look like Discord would investigate every single rate limit exceeded so I should be okay running this. :+1:
